lfsapi/auth.go

```diff
@@ -24,39 +24,53 @@ var (
 // DoWithAuth sends an HTTP request to get an HTTP response. It attempts to add
 // authentication from netrc or git's credential helpers if necessary,
 // supporting basic and ntlm authentication.
-func (c *Client) DoWithAuth(remote string, req *http.Request) (*http.Response, error) {
-	return c.doWithAuth(remote, req, nil)
+func (c *Client) DoWithAuth(remote string, access Access, req *http.Request, allowRetry bool) (*http.Response, error) {
+	res, err := c.doWithAuth(remote, access, req, nil)
+
+	if allowRetry && errors.IsAuthError(err) {
+		if len(req.Header.Get("Authorization")) == 0 {
```

Comment:
This is a change to the retry conditional that previously lived in doWithAuth(), but it seemed safe to me. If a request fails with credentials then the Authorization header will be deleted before doWithAuth() returns, so this should cover that case, and so should only fire if the request had no authorization or if it used a credential helper to fill out the Authorization header.